src/test/java/dog/pawbook/logic/commands/EditOwnerDescriptorTest.java

```diff
@@ -36,7 +36,8 @@ public void equals() {
         assertFalse(DESC_AMY.equals(DESC_BOB));
 
         // different name -> returns false
-        EditOwnerDescriptor editedAmy = new EditOwnerDescriptorBuilder(DESC_AMY).withName(VALID_NAME_BOB).build();
+        EditOwnerDescriptor editedAmy = (EditOwnerDescriptor) new EditOwnerDescriptorBuilder(DESC_AMY)
```

Comment:
do you mind explaining why casting is necessary?

Reply:
I just copy n paste from the previous PR. This part was accepted last time so I assume it works for now?

Reply:
I don't want to do any finger-pointing here, while this is my oversight the previous time, it would be helpful if you broke down your commits such that every commit has a purposeful message along with the stated changes, that way it is easier for me look through at each step and figure out if anything is wrong. However, all that I get here is around 1k lines of code changed, barring the fact that the previous commit history didn't carry over, it is really difficult to be that careful. I won't request that you review what I changed for you in the previous PR, in short your builder pattern was less than ideal, thus sub-type information was lost, hence requiring the casting. I fixed it to avoid those issues, but since extra casting is not technically wrong but just poor coding practice, there wasn't anything that alerted or reminded me that these portions needed to be updated to remove the casting. The fix here is simple, just remove the cast. Do let me know if you don't understand any part.

Reply:
Remove!

Reply:
It is not just whether or not someone is going to review it, git commits are supposed to be incremental regardless, the previous PR could have been kept by merging in the latest master and resolving the conflicts. By creating another PR the commit history and authorship are all gone, that's the same reason I previously tried to restrain myself from directly pushing to existing PRs and make everyone perform the minor changes themselves, it actually takes more effort to type all the comments I have given than to directly modify them myself.